### PR TITLE
Keys to the castle

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
## About The Pull Request
Janitor keys - new item for janitor and only janitors to open almost any non-bolted door
This comes with the new perk to make sure were not abusing it.
Different doors have different odds.

## Changelog
:cl:
/:cl:

### CODE REVIEW

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -977,6 +1012,32 @@ There are 9 wires.
...
+				playsound(src.loc, 'sound/items/keychainrattle.ogg', 30, 1, -2)
+			else
+				playsound(src.loc, 'sound/items/keychainrattle.ogg', 700, 1, -2)
+			if(do_after(user, 900, src))
+				used_now = FALSE
+				if(locked)
+					to_chat(user, SPAN_NOTICE("Even with the right key you can't open \"deadbolts\"!"))
+					used_now = FALSE
+				return
+				if(prob(key_odds+1) && H.stats.getPerk(PERK_ARTIST)) //minmium 1%
```
Comment (Kazkin) [inline code/game/machinery/doors/airlock.dm:1032]: Perk artist?

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -977,6 +1012,32 @@ There are 9 wires.
...
+		used_now = TRUE
+		if(ishuman(usr))
+			var/mob/living/carbon/human/H = usr
+			if(istype(I, /obj/item/weapon/keys/lockpicks))
+				playsound(src.loc, 'sound/items/keychainrattle.ogg', 30, 1, -2)
+			else
+				playsound(src.loc, 'sound/items/keychainrattle.ogg', 700, 1, -2)
+			if(do_after(user, 900, src))
+				used_now = FALSE
+				if(locked)
```
Comment (Kazkin) [inline code/game/machinery/doors/airlock.dm:1028]: It should check if the door is locked before it starts the timer, earlier return failure should always be first.

Comment (Kazkin) [reply code/game/machinery/doors/airlock.dm:1028]: It should also check if the door has power or not.

Comment (Trilbyspaceclone) [reply code/game/machinery/doors/airlock.dm:1028]: If it has no power then it dosnt open as its not forced!
Also its a 100% intented to not have it check before the bolt as you can hide away bolt light giving janitors a soft-check rather then a hard check

**Pull request merged**